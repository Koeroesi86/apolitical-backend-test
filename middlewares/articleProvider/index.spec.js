const articleProviderMiddleware = require('./index');
const { createRequest, createResponse } = require("node-mocks-http");

const provider = jest.fn(() => Promise.resolve('Lorem ipsum'));

describe('articleProvider middleware', () => {
  const middleware = articleProviderMiddleware(provider);

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should respond with article', async () => {
    const request = createRequest();
    const response = createResponse();
    const next = jest.fn();

    await middleware(request, response, next);

    expect(next).toHaveBeenCalledTimes(0);
    expect(request).toMatchSnapshot('request');
    expect(response).toMatchSnapshot('response');
  });

  it('should call next on error', async () => {
    const request = createRequest();
    const response = createResponse();
    const next = jest.fn();
    provider.mockRejectedValueOnce(new Error('example'))

    await middleware(request, response, next);

    expect(next).toHaveBeenCalled();
    expect(next.mock.calls).toMatchSnapshot();
  });
})
