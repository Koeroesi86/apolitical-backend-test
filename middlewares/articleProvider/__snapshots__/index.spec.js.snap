// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleProvider middleware should call next on error 1`] = `
Array [
  Array [
    [Error: example],
  ],
]
`;

exports[`articleProvider middleware should respond with article: request 1`] = `
EventEmitter {
  "_addBody": [Function],
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "_setBaseUrl": [Function],
  "_setBody": [Function],
  "_setCookiesVariable": [Function],
  "_setFilesVariable": [Function],
  "_setHeadersVariable": [Function],
  "_setMethod": [Function],
  "_setOriginalUrl": [Function],
  "_setParameter": [Function],
  "_setSessionVariable": [Function],
  "_setSignedCookiesVariable": [Function],
  "_setURL": [Function],
  "accepts": [Function],
  "acceptsCharsets": [Function],
  "acceptsEncodings": [Function],
  "acceptsLanguages": [Function],
  "baseUrl": "",
  "body": Object {},
  "cookies": Object {},
  "files": Object {},
  "get": [Function],
  "header": [Function],
  "headers": Object {},
  "hostname": "",
  "is": [Function],
  "method": "GET",
  "originalUrl": "",
  "param": [Function],
  "params": Object {},
  "path": "",
  "query": Object {},
  "range": [Function],
  "send": [Function],
  "socket": Object {},
  "subdomains": Array [],
  "url": "",
  Symbol(kCapture): false,
}
`;

exports[`articleProvider middleware should respond with article: response 1`] = `
EventEmitter {
  "_getBuffer": [Function],
  "_getChunks": [Function],
  "_getData": [Function],
  "_getHeaders": [Function],
  "_getJSONData": [Function],
  "_getLocals": [Function],
  "_getRedirectUrl": [Function],
  "_getRenderData": [Function],
  "_getRenderView": [Function],
  "_getStatusCode": [Function],
  "_getStatusMessage": [Function],
  "_headers": Object {},
  "_isDataLengthValid": [Function],
  "_isEndCalled": [Function],
  "_isJSON": [Function],
  "_isUTF8": [Function],
  "append": [Function],
  "clearCookie": [Function],
  "contentType": [Function],
  "cookie": [Function],
  "cookies": Object {},
  "destroy": [Function],
  "destroySoon": [Function],
  "end": [Function],
  "finished": true,
  "format": [Function],
  "get": [Function],
  "getEncoding": [Function],
  "getHeader": [Function],
  "getHeaderNames": [Function],
  "getHeaders": [Function],
  "hasHeader": [Function],
  "header": [Function],
  "headersSent": true,
  "json": [Function],
  "jsonp": [Function],
  "locals": Object {},
  "location": [Function],
  "redirect": [Function],
  "removeHeader": [Function],
  "render": [Function],
  "send": [Function],
  "sendStatus": [Function],
  "set": [Function],
  "setEncoding": [Function],
  "setHeader": [Function],
  "status": [Function],
  "statusCode": 200,
  "statusMessage": "OK",
  "type": [Function],
  "vary": [Function],
  "writableEnded": true,
  "writableFinished": true,
  "write": [Function],
  "writeHead": [Function],
}
`;
